core/src/main/java/org/bitcoinj/core/Wallet.java

```diff
@@ -3971,7 +3971,8 @@ public boolean canSignFor(Script script) {
     protected List<UTXO> getStoredOutputsFromUTXOProvider() throws UTXOProviderException {
         UTXOProvider utxoProvider = checkNotNull(vUTXOProvider, "No UTXO provider has been set");
         List<UTXO> candidates = new ArrayList<UTXO>();
-        List<DeterministicKey> keys = getActiveKeyChain().getLeafKeys();
+        List<ECKey> keys = getImportedKeys();
+        keys.addAll(getActiveKeyChain().getLeafKeys());
```

Comment:
@schildbach ah I see, the getImportedKeys method calls the basic.getKeys() and this method returns a **copy** of the imported keys list:

BasicKeyChain.java getKeys() returns this:
```
return new ArrayList<ECKey>(hashToKeys.values());
```

So I think it's safe to modify this returned list.